File: csharp/collatz-conjecture/CollatzConjecture.cs
using System;

public static class CollatzConjecture
{
    public static int Steps(int number)
    {
        if (number <= 0) throw new ArgumentException("Integer has to be positive");

        int steps = 0;
        while (number > 1)
        {
            steps++;
            if (number % 2 == 0) number /= 2;
            else number = 3 * number + 1;
        }

        return steps;
    }
}

File: go/collatz-conjecture/collatz_conjecture.go
package collatzconjecture

import "errors"

func CollatzConjecture(n int) (int, error) {
	if n < 1 {
		return -1, errors.New("Only positive numbers allowed.")
	}

	steps := 0

	for n != 1 {
		steps += 1

		if n%2 == 0 {
			n = n / 2
		} else {
			n = 3*n + 1
		}
	}

	return steps, nil
}
